lib/kubernetes-deploy/kubernetes_resource/replica_set.rb

```diff
@@ -53,7 +61,8 @@ def exists?
     def fetch_events
       own_events = super
       return own_events unless @pods.present?
-      own_events.merge(@pods.first.fetch_events)
+      most_useful_pod = @pods.find(&:deploy_failed?) || @pods.find(&:deploy_timed_out?) || @pods.first
```

Comment:
Yes, exactly. If there aren't any failed/timed out pods, all pods should be pretty much equal for debugging purposes. The case where we'll definitely hit the `pods.first` fallback is where the timeout was triggered by the parent, not a pod.